akka-actor/src/main/scala/akka/actor/FaultHandling.scala

```diff
@@ -155,6 +155,7 @@ object SupervisorStrategy extends SupervisorStrategyLowPriorityImplicits {
     case _: ActorInitializationException ⇒ Stop
     case _: ActorKilledException         ⇒ Stop
     case _: DeathPactException           ⇒ Stop
+    case m: IncompatibleClassChangeError ⇒ detectAkkaBinaryIncompatibilityWarningAndStop(m)
```

Comment:
Though linking is lazy in the jvm so it could happen the first time someone touches some API as well... The suggestion to move it to actor system is a good one though, will follow up on that!